File: code/src/StandardBank.ConcessionManagement.UI/client-src/app/models/trade-concession-detail.ts
import { BaseConcessionDetail } from "./base-concession-detail";

export class TradeConcessionDetail extends BaseConcessionDetail {
    tradeConcessionDetailId: number;

    legalEntity: string;   
  
    tradeProduct: string;
    tradeProductType: string;
    approvedRate: number;
    loadedRate :number;
    communication: string;
    establishmentFee: number;
    flatFee: number;
    gbbNumber: string;
    max: number;
    min: number;
    term: number;
    fkTradeProductId: number;
    tradeProductTypeID: number;

    fkLegalEntityAccountId: number;  
    fkLegalEntityGBBNumber: number;

    adValorem: number;
    currency: number;

    show_term: boolean;
    show_advalorem: boolean;
    show_min: boolean;
    show_max: boolean;
    show_communication: boolean;
    show_flatfee: boolean;

    disablecontrolset: boolean;
}

File: code/src/StandardBank.ConcessionManagement.Model/UserInterface/Trade/TradeConcessionDetail.cs
namespace StandardBank.ConcessionManagement.Model.UserInterface.Trade
{
    /// <summary>
    /// Cash concession detail entity
    /// </summary>
    public class TradeConcessionDetail : BaseConcessionDetail
    {
        public int TradeConcessionDetailId { get; set; }
        public string LegalEntity { get; set; }
        public string TradeProduct { get; set; }
        public string TradeProductType { get; set; } 
        public string ApprovedRate { get; set; }
        public string LoadedRate { get; set; }
        public string Rate { get; set; }
        public int? fkConcessionId { get; set; }       
        public int? fkConcessionDetailId { get; set; }
        public string Communication { get; set; }
        public double EstablishmentFee { get; set; }
        public double FlatFee { get; set; }
        public string GBBNumber { get; set; }
        public double max { get; set; }
        public double min { get; set; }
        public int term { get; set; }        

        public int? fkTradeProductId { get; set; }
       
        public int? fkLegalEntityAccountId { get; set; }

        public int? fkLegalEntityGBBNumber { get; set; }

        public int? fkLegalEntityId { get; set; }

        public decimal? AdValorem { get; set; }

        public string Currency { get; set; }





    }
}
